src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -995,20 +1004,142 @@ static inline JSC::EncodedJSValue jsBufferPrototypeFunction_fillBody(JSC::JSGlob
     }
 }
 
-static inline JSC::EncodedJSValue jsBufferPrototypeFunction_includesBody(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame, typename IDLOperation<JSBuffer>::ClassParameter castedThis)
+static int indexOf(const uint8_t* thisPtr, size_t thisLength, const uint8_t* valuePtr, size_t valueLength, int byteOffset)
+{
+    if (thisLength < valueLength + byteOffset)
+        return -1;
+    auto start = thisPtr + byteOffset;
+    auto end = thisPtr + thisLength;
+    auto it = std::search(start, end, std::boyer_moore_searcher(valuePtr, valuePtr + valueLength));
```

Comment:
lets use `memmem` here

C++ stdlib tends to perform worse than C stdlib

Reply:
fixed.